From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "p.tumakaev" <p.tumakaev@omprussia.ru>
Date: Mon, 29 Jun 2020 14:29:52 +0300
Subject: [PATCH] [sailfishos][gecko] Skip invalid WatchId in
 geolocation.clearWatch(). Contributes to JB#50110

ClearWatch() stores passed WatchId in mClearedWatchIDs without checking
if it was previously generated by WatchPosition(). As a result, all
subsequent WatchPosition() with such invalid identifiers are automatically
canceled at the "Allow" stage.

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 dom/geolocation/Geolocation.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dom/geolocation/Geolocation.cpp b/dom/geolocation/Geolocation.cpp
index 2477e5189e0d..de4ccf68d9f8 100644
--- a/dom/geolocation/Geolocation.cpp
+++ b/dom/geolocation/Geolocation.cpp
@@ -1146,7 +1146,7 @@ int32_t Geolocation::WatchPosition(GeoPositionCallback aCallback,
 }
 
 void Geolocation::ClearWatch(int32_t aWatchId) {
-  if (aWatchId < 0) {
+  if (aWatchId < 0 || aWatchId >= mLastWatchId) {
     return;
   }
 
-- 
2.31.1

